### PR TITLE
docs: add architecture background

### PR DESCRIPTION
**Pull request by bbangert**
This is intended to provide sufficient documentation about what this does, and how the process works. So feel free to be harsh on more details needed, etc.

@jrconlin r?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,26 @@ A [swagger](http://swagger.io/) driven Pyramid app that utilizes Redis for
 recent messages sent into the Mozilla Push System and manages crypto public keys
 that should have their recent message history recorded.
 
+## Architecture
+
+This project provides an internal API for the Mozilla Push Developer Dashboard
+to maintain a set of VAPID public keys that are registered developers so that
+a separate process can populate the Redis database used.
+
```
Comment (jrconlin) [inline README.md:14]: That reads a bit hard. 

Is this better? 

> This project provides an internal API for the Mozilla Push Developer Dashboard. Developers are identified by their registered VAPID public keys. A separate process populates those keys to a Redis database.

File: README.md [hunk 2]
```diff
@@ -6,6 +6,26 @@ A [swagger](http://swagger.io/) driven Pyramid app that utilizes Redis for
...
+metadata extracted from the Mozilla Push Service logstream.
+
+This is a very lightly used table managed by `push_messages`.
+
+### Redis
+
+Stores a circular list for each VAPID public key of recent message metadata
+that `push_messages` can retrieve via API calls. Redis is used as it supports
+efficient access/retrieval and has trim operations to emulate a circular list
+structure efficiently.
```
Comment (jrconlin) [inline README.md]: sorry, but does: 

> Stores a circular list of recent message metadata, indexed by VAPID public key. `push_messages` can retrieve this data via API calls. Redis supports efficient access/retrieval and has trim operations to emulate a circular list structure efficiently.

read better?

File: README.md (hunk 1 above)
Comment (bbangert) [inline README.md:14]: Definitely.

File: README.md (hunk 2 above)
Comment (bbangert) [inline README.md]: yup!

**Comment by jrconlin** [comment]
r+

**Pull request merged**